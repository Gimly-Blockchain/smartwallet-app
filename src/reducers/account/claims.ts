import { AnyAction } from 'redux'
import {
  ClaimsState,
  CategorizedClaims,
  DecoratedClaims,
} from 'src/reducers/account'

const categorizedClaims: CategorizedClaims = {
  Personal: [
    {
      credentialType: 'Name',
      claimData: {
        givenName: '',
        familyName: '',
      },
      id: '',
      issuer: {
        did: '',
      },
      subject: '',
    },
  ],
  Contact: [
    {
      credentialType: 'Email',
      claimData: {
        email: '',
      },
      id: '',
      issuer: {
        did: '',
      },
      subject: '',
      keyboardType: 'email-address',
    },
    {
      credentialType: 'Mobile Phone',
      claimData: {
        telephone: '',
      },
      id: '',
      issuer: {
        did: '',
      },
      subject: '',
      keyboardType: 'phone-pad',
    },
    {
      credentialType: 'Postal Address',
      claimData: {
        addressLine1: '',
        addressLine2: '',
        postalCode: '',
        city: '',
        country: '',
      },
      id: '',
      issuer: {
        did: '',
      },
      subject: '',
    },
  ],
  // /** @dev FOR TESTING */
  Other: [],
}

export const initialState: ClaimsState = {
  loading: false,
  selected: {
    credentialType: '',
    claimData: {},
    id: '',
    issuer: {
      did: '',
    },
    subject: '',
  },
  pendingExternal: {
    offeror: {
      did: '',
    },
    offer: [],
  },
  decoratedCredentials: categorizedClaims,
}

export const claims = (
  state = initialState,
  action: AnyAction,
): ClaimsState => {
  switch (action.type) {
    case 'TOGGLE_CLAIMS_LOADING':
      return { ...state, loading: action.value }
    case 'SET_CLAIMS_FOR_DID':
      return { ...state, decoratedCredentials: addDefaultValues(action.claims) }
    case 'SET_EXTERNAL':
      return { ...state, pendingExternal: action.value }
    case 'RESET_EXTERNAL':
      return { ...state, pendingExternal: initialState.pendingExternal } // TODO Remove in favor of calling set external with empty array
    case 'SET_SELECTED':
      return { ...state, selected: action.selected }
    case 'RESET_SELECTED':
      return { ...state, selected: initialState.selected }
    case 'HANDLE_CLAIM_INPUT':
      return {
        ...state,
        selected: {
          ...state.selected,
          claimData: {
            ...state.selected.claimData,
            [action.fieldName]: action.fieldValue,
          },
        },
      }
    default:
      return state
  }
}

const addDefaultValues = (claims: CategorizedClaims) =>
  Object.keys(categorizedClaims).reduce(
    (acc: CategorizedClaims, category: string) => ({
      ...acc,
      [category]: injectPlaceholdersIfNeeded(category, claims[category]),
    }),
    {},
  )

const injectPlaceholdersIfNeeded = (
  category: string,
  claims: DecoratedClaims[],
): DecoratedClaims[] => {
  if (!claims || claims.length === 0) {
    return categorizedClaims[category]
  }

  const missing = categorizedClaims[category].filter(
    defaultClaim =>
      !claims.some(
        claim => claim.credentialType === defaultClaim.credentialType,
      ),
  )

  if (missing) {
    return [...missing, ...claims]
  }

  return claims
}
