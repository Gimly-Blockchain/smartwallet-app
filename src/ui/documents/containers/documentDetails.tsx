import React from 'react'
import { connect } from 'react-redux'
import { NavigationScreenProps } from 'react-navigation'
import { View, StyleSheet, ScrollView } from 'react-native'
import { ClaimInterface } from 'cred-types-jolocom-core'
import { RootState } from 'src/reducers/'
import I18n from 'src/locales/i18n'
import { DocumentCard } from '../components/documentCard'
import { DocumentDetails as DocumentDetailsComponent } from '../components/documentDetails'
import strings from '../../../locales/strings'
import { JolocomTheme } from 'src/styles/jolocom-theme.ios'

interface Props
  extends ReturnType<typeof mapStateToProps>,
    NavigationScreenProps {}

interface State {}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: JolocomTheme.primaryColorGrey,
  },
  documentCardContainer: {
    paddingVertical: 15,
    justifyContent: 'center',
    alignItems: 'center',
  },
})

export class DocumentDetailsContainer extends React.Component<Props, State> {
  public static navigationOptions = ({
    navigation,
    navigationOptions,
  }: NavigationScreenProps) => ({
    ...navigationOptions,
    // FIXME needs to be changed to this on update to react-navigation
    // headerTitle: navigation.getParam('headerTitle', ''),
    headerTitle: navigation.state.params && navigation.state.params.headerTitle,
  })

  public componentWillMount() {
    const { document } = this.props
    // FIXME use filter from src/lib/filterDecoratedClaims
    const isExpired = document.expires
      ? document.expires.valueOf() < new Date().valueOf()
      : true

    let headerTitle = (document.claimData as ClaimInterface).type
    if (isExpired) headerTitle = `[${I18n.t(strings.EXPIRED)}] ${headerTitle}`
    this.props.navigation.setParams({ headerTitle })
  }

  public render(): JSX.Element {
    const { document } = this.props

    return (
      <View style={styles.container}>
        <View style={styles.documentCardContainer}>
          <DocumentCard document={document} />
        </View>
        <ScrollView>
          <DocumentDetailsComponent document={document} />
        </ScrollView>
      </View>
    )
  }
}

const mapStateToProps = (state: RootState) => ({
  document: state.documents.selectedDocument,
})

export const DocumentDetails = connect(mapStateToProps)(
  DocumentDetailsContainer,
)
