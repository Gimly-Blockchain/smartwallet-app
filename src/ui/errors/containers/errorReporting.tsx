import React, { useState } from 'react'
import { connect } from 'react-redux'
import { NavigationScreenProp, NavigationState } from 'react-navigation'
import { AppError, reportError } from '../../../lib/errors'
import { ThunkDispatch } from '../../../store'
import { routeList } from '../../../routeList'
import { navigationActions } from '../../../actions'
import { Emoji, EmojiSection } from '../components/emojiSection'
import { Container } from '../../structure'
import { styles } from '../styles'
import { ScrollView } from 'react-native'
import { ChooseIssueSection } from '../components/chooseIssueSection'
import { DescriptionSection } from '../components/descriptionSection'
import { ContactSection } from '../components/contactSection'
import { GradientButton } from '../../structure/gradientButton'
import I18n from '../../../locales/i18n'
import strings from '../../../locales/strings'
interface PaymentNavigationParams {
  error: AppError | Error | undefined
}

interface Props extends ReturnType<typeof mapDispatchToProps> {
  navigation: NavigationScreenProp<NavigationState, PaymentNavigationParams>
}

export enum Inputs {
  None,
  Dropdown,
  Description,
  Contact,
}

const ErrorReportingContainer = (props: Props) => {
  const { navigateToScreen, navigation } = props

  const [pickedIssue, setIssue] = useState<string>()
  const [description, setDescription] = useState<string>('')
  const [contact, setContact] = useState<string>('')
  const [currentInput, setInput] = useState<Inputs>(Inputs.None)
  const [toggleState, setToggle] = useState<boolean>(false)
  const [selectedEmoji, setEmoji] = useState<Emoji>(Emoji.Empty)

  const onSubmitReport = () => {
    const { error } = props.navigation.state.params

    const userReport = {
      userError: pickedIssue,
      userDescription: description,
      userContact: contact,
    }
    console.log(userReport)

    if (navigation && error) {
      reportError({ ...userReport, error })
      if (error instanceof AppError) {
        navigateToScreen(error.navigateTo)
      } else {
        navigateToScreen(routeList.AppInit)
      }
    }
  }

  return (
    <Container style={styles.wrapper}>
      <ScrollView>
        <ChooseIssueSection
          currentInput={currentInput}
          pickedIssue={pickedIssue}
          onIssuePick={setIssue}
          setInput={setInput}
        />
        <DescriptionSection
          currentInput={currentInput}
          setInput={setInput}
          setDescription={setDescription}
          toggleState={toggleState}
          setToggle={setToggle}
          description={description}
        />
        <ContactSection
          onContactInput={setContact}
          currentInput={currentInput}
          setInput={setInput}
          contactValue={contact}
        />
        <EmojiSection selectedEmoji={selectedEmoji} setEmoji={setEmoji} />
        <GradientButton
          onPress={onSubmitReport}
          containerStyle={{
            marginTop: 45,
            marginBottom: 66,
          }}
          text={I18n.t(strings.SUBMIT_REPORT)}
        />
      </ScrollView>
    </Container>
  )
}

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  navigateToScreen: (screen: routeList) =>
    dispatch(navigationActions.navigate({ routeName: screen })),
})

export const ErrorReporting = connect(
  null,
  mapDispatchToProps,
)(ErrorReportingContainer)
