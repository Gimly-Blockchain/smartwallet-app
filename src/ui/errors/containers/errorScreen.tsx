import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { JolocomButton, Wrapper } from '../../structure'
import { NavigationScreenProp, NavigationState } from 'react-navigation'
import { centeredText, fontMain } from '../../../styles/typography'
import { Colors } from '../../../styles'
import { BP } from '../../../styles/breakpoints'
import { BlueCircleImage, RedCircleImage } from '../../../resources'

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: Colors.overflowBlack,
    justifyContent: 'space-between',
  },
  imageWrapper: {
    marginTop: '15%',
    width: '65%',
    height: '54%',
  },
  title: {
    paddingHorizontal: 20,
    fontFamily: fontMain,
    fontSize: BP({
      small: 26,
      medium: 28,
      large: 28,
    }),
    color: Colors.sandLight090,
    ...centeredText,
  },
  message: {
    marginTop: 13,
    paddingHorizontal: 20,
    fontFamily: fontMain,
    fontSize: BP({
      small: 18,
      medium: 20,
      large: 20,
    }),
    color: Colors.sandLight080,
    lineHeight: 24,
    ...centeredText,
  },
  interactButton: {
    marginTop: 14,
    borderColor: 'white',
    borderWidth: 0.5,
  },
  buttonSection: {
    width: '100%',
    paddingBottom: 62,
    paddingHorizontal: 20,
    justifyContent: 'space-between',
  },
  contentSection: {
    width: '100%',
    alignItems: 'center',
  },
})

enum ImageType {
  Red = 'red',
  Blue = 'blue',
}

export interface ErrorScreenParams {
  title: string
  message: string
  image?: ImageType
  interact?: {
    label: string
    onInteract: () => void | Promise<void>
  }
  dismiss: {
    label: string
    onDismiss: () => void | Promise<void>
  }
}

interface Props {
  navigation: NavigationScreenProp<NavigationState, ErrorScreenParams>
}

export const ErrorScreenContainer: React.FC<Props> = props => {
  const {
    navigation: {
      state: {
        params: { title, message, interact, dismiss, image },
      },
    },
  } = props

  return (
    <Wrapper style={styles.wrapper}>
      <View style={styles.contentSection}>
        <View style={styles.imageWrapper}>
          {image === ImageType.Red ? <RedCircleImage /> : <BlueCircleImage />}
        </View>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.message}>{message}</Text>
      </View>
      <View style={styles.buttonSection}>
        <JolocomButton
          onPress={dismiss.onDismiss}
          transparent
          text={dismiss.label}
        />
        {interact && (
          <JolocomButton
            containerStyle={styles.interactButton}
            transparent
            onPress={interact.onInteract}
            text={interact.label}
          />
        )}
      </View>
    </Wrapper>
  )
}
