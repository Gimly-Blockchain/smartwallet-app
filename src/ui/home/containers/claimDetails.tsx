import React from 'react'
import { ClaimDetailsComponent } from 'src/ui/home/components/claimDetails'
import { connect } from 'react-redux'
import { RootState } from 'src/reducers/'
import { accountActions } from 'src/actions'
import { ThunkDispatch } from '../../../store'
import { withErrorHandling, withLoading } from '../../../actions/modifiers'
import { showErrorScreen } from '../../../actions/generic'
import { setDeepLinkLoading } from '../../../actions/sso'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps> {}

interface State {}

export class ClaimDetailsContainer extends React.Component<Props, State> {
  render() {
    return (
      <ClaimDetailsComponent
        saveClaim={this.props.saveClaim}
        handleClaimInput={this.props.handleClaimInput}
        selectedClaim={this.props.claims.selected}
      />
    )
  }
}

const mapStateToProps = ({ account: { claims } }: RootState) => ({
  claims,
})

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  saveClaim: () =>
    dispatch(
      withLoading(setDeepLinkLoading)(
        withErrorHandling(showErrorScreen)(accountActions.saveClaim),
      ),
    ),
  handleClaimInput: (fieldValue: string, fieldName: string) => {
    dispatch(accountActions.handleClaimInput(fieldValue, fieldName))
  },
})

export const ClaimDetails = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ClaimDetailsContainer)
