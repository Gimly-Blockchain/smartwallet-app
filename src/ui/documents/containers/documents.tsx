import React from 'react'
import { connect } from 'react-redux'
import { Animated, ScrollView, StyleSheet, Text, View } from 'react-native'

import { openDocumentDetails } from 'src/actions/documents'
import { DecoratedClaims } from 'src/reducers/account'
import { RootState } from 'src/reducers'
import { getDocumentClaims } from 'src/utils/filterDocuments'
import { ThunkDispatch } from 'src/store'

import I18n from 'src/locales/i18n'
import { filters } from 'src/lib/filterDecoratedClaims'

import { DocumentsCarousel } from '../components/documentsCarousel'
import { DocumentsList } from '../components/documentsList'
import { DocumentViewToggle } from '../components/documentViewToggle'
import strings from '../../../locales/strings'
import { Typography, Colors, Spacing } from 'src/styles'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps> {}

interface State {
  activeDocumentIndex: number
  showingValid: boolean
}

/*
const APPBAR_HEIGHT = Platform.select({
  ios: 44,
  android: 56,
  default: 64,
})
*/

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: Colors.lightGreyLighter,
  },
  topContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: Spacing.MD,
  },
  emptyDocumentsContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: Spacing.MD,
  },
  emptyDocumentsText: {
    ...Typography.mainText,
    ...Typography.centeredText,
    color: Colors.greyLight,
  },
})

export class DocumentsContainer extends React.Component<Props, State> {
  public ScrollViewRef: ScrollView | null = null

  public state = {
    activeDocumentIndex: 0,
    showingValid: true,
  }

  private scrollToTop(): void {
    if (this.ScrollViewRef) {
      this.ScrollViewRef.scrollTo({ x: 0, y: 0, animated: true })
    }
  }

  private onActiveIndexChange(index: number): void {
    this.setState({ activeDocumentIndex: index })
    this.scrollToTop()
  }

  private handleToggle = () => {
    this.setState((prevState: State) => ({
      showingValid: !prevState.showingValid,
      activeDocumentIndex: 0,
    }))
    this.scrollToTop()
  }

  public render(): JSX.Element {
    const { openDocumentDetails, decoratedCredentials } = this.props
    const documents = getDocumentClaims(decoratedCredentials['Other'])
    const docFilter = this.state.showingValid
      ? filters.filterByValid
      : filters.filterByExpired
    const displayedDocs = docFilter(documents)
    const isEmpty = displayedDocs.length === 0
    const otherIsEmpty = displayedDocs.length === documents.length

    return (
      <Animated.View style={styles.mainContainer}>
        {!otherIsEmpty && (
          <DocumentViewToggle
            showingValid={this.state.showingValid}
            onTouch={this.handleToggle}
          />
        )}
        <ScrollView
          // scrollEventThrottle={16}
          // onScroll={Animated.event([
          //   {
          //     nativeEvent: {
          //       contentOffset: { y: this.state.headerHeight },
          //     },
          //   },
          // ])}
          // to scroll to top upon changing card
          ref={ref => (this.ScrollViewRef = ref)}
          contentContainerStyle={isEmpty && { flex: 1 }}
          scrollEnabled={!isEmpty}
        >
          {isEmpty ? (
            <View style={styles.emptyDocumentsContainer}>
              <Text style={styles.emptyDocumentsText}>
                {I18n.t(strings.NO_DOCUMENTS_TO_SEE_HERE) + '...'}
              </Text>
            </View>
          ) : this.state.showingValid ? (
            <DocumentsCarousel
              documents={displayedDocs}
              activeIndex={this.state.activeDocumentIndex}
              onActiveIndexChange={this.onActiveIndexChange.bind(this)}
            />
          ) : (
            <DocumentsList
              documents={displayedDocs}
              onDocumentPress={openDocumentDetails}
            />
          )}
        </ScrollView>
      </Animated.View>
    )
  }
}

const mapStateToProps = ({
  account: {
    claims: { decoratedCredentials },
  },
}: RootState) => ({
  decoratedCredentials,
})

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  openDocumentDetails: (doc: DecoratedClaims) =>
    dispatch(openDocumentDetails(doc)),
})

export const Documents = connect(
  mapStateToProps,
  mapDispatchToProps,
)(DocumentsContainer)
