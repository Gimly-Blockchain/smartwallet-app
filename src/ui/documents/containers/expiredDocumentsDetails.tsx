import React from 'react'
import { ExpiredDocumentsDetailsComponent } from 'src/ui/documents/components/expiredDocumentDetails'
import { connect } from 'react-redux'
import { RootState } from 'src/reducers/'
import { NavigationScreenProps } from 'react-navigation'
import { DecoratedClaims } from 'src/reducers/account'

interface ConnectProps {
  selectedExpiredDocument: DecoratedClaims
}

interface Props extends ConnectProps {
  documents: Document[]
}

interface State {}

export class ExpiredDocumentsDetailsContainer extends React.Component<
  Props,
  State
> {
  public static navigationOptions = ({
    navigation,
    navigationOptions,
  }: NavigationScreenProps) => ({
    ...navigationOptions,
    headerTitle: navigation.getParam('headerTitle', 'DEFAULT'),
  })

  public render(): JSX.Element {
    const { selectedExpiredDocument } = this.props
    return (
      <ExpiredDocumentsDetailsComponent document={selectedExpiredDocument} />
    )
  }
}

const mapStateToProps = (state: RootState): {} => ({
  selectedExpiredDocument: state.documents.selectedExpiredDocument,
})

const mapDispatchToProps = (dispatch: Function): {} => ({})

export const ExpiredDocumentsDetails = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ExpiredDocumentsDetailsContainer)
