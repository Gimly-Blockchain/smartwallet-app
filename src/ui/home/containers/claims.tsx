import React from "react"
import { CredentialOverview } from "../components/credentialOverview"
import { QRcodeScanner } from "src/ui/home/components/qrcodeScanner"
import { connect } from "react-redux"
import { accountActions, ssoActions, navigationActions } from "src/actions"
import { View } from "react-native"
import { ClaimsState } from "src/reducers/account"
import { DecoratedClaims } from "src/reducers/account/"
import { QrScanEvent } from "./types"
import { LoadingSpinner } from "../../generic"
import { routeList } from "src/routeList"

interface ConnectProps {
  setClaimsForDid: () => void
  toggleLoading: (val: boolean) => void
  parseJWT: (jwt: string) => void
  openClaimDetails: (claim: DecoratedClaims) => void
  openScanner: () => void
  did: string
  claims: ClaimsState
  loading: boolean
}

interface Props extends ConnectProps {}

interface State {
  scanning: boolean
  loading: boolean
}

export class ClaimsContainer extends React.Component<Props, State> {
  state = {
    scanning: false,
    loading: false
  }

  componentWillMount() {
    this.props.setClaimsForDid()
  }

  private onScannerStart = (): void => {}

  private onScannerCancel = (): void => {
    this.setState({ scanning: false })
    this.setState({ loading: false })
  }

  private onScannerSuccess = (e: QrScanEvent): void => {
    this.setState({ scanning: false })
    this.props.parseJWT(e.data)
    this.setState({ loading: false })
  }

  render() {
    if (this.state.scanning) {
      return (
        <QRcodeScanner
          onScannerSuccess={this.onScannerSuccess}
          onScannerCancel={this.onScannerCancel}
        />
      )
    }

    if (this.state.loading || this.props.loading || this.props.claims.loading) {
      return <LoadingSpinner />
    }
    return (
      <View style={{ flex: 1 }}>
        <CredentialOverview
          did={this.props.did}
          claimsState={this.props.claims}
          loading={this.props.claims.loading}
          onEdit={this.props.openClaimDetails}
          scanning={this.state.scanning}
          onScannerStart={this.onScannerStart}
        />
      </View>
    )
  }
}

// TODO nicer pattern for accessing state, perhaps immer or something easier to Type
const mapStateToProps = (state: any) => {
  return {
    did: state.account.did.toJS().did,
    claims: state.account.claims.toJS(),
    loading: state.account.loading.toJS().loading
  }
}

const mapDispatchToProps = (dispatch: Function) => {
  return {
    openClaimDetails: (claim: DecoratedClaims) =>
      dispatch(accountActions.openClaimDetails(claim)),
    setClaimsForDid: () => dispatch(accountActions.setClaimsForDid()),
    toggleLoading: (val: boolean) =>
      dispatch(accountActions.toggleLoading(val)),
    parseJWT: (jwt: string) => dispatch(ssoActions.parseJWT(jwt)),
    openScanner: () =>
      dispatch(
        navigationActions.navigate({ routeName: routeList.QRCodeScanner })
      )
  }
}

export const Claims = connect(
  mapStateToProps,
  mapDispatchToProps
)(ClaimsContainer)
