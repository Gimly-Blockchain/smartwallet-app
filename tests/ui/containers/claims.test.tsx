import React from 'react'
import { shallow } from 'enzyme'
import { ClaimsContainer } from 'src/ui/home/containers/claims'


describe('Claims container', () => {
  const COMMON_PROPS = {
    scanning: false,
    showClaimsDetails: false,
    typeClaimDetails: '',
    claims: {},
    toggleLoading: () => {}
  }

  it('mounts correctly and matches the snapshot', () => {
    const setClaimsForDid = jest.fn()

    const props = Object.assign({}, COMMON_PROPS {
      setClaimsForDid
    })

    const rendered = shallow(<ClaimsContainer {...props}/>)

    expect(rendered).toMatchSnapshot()
    expect(setClaimsForDid).toHaveBeenCalledTimes(1)
    })
})
