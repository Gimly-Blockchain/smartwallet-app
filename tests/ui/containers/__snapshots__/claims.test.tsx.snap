// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Claims container correctly changes scanning to false when qr code scanner is canceled 1`] = `
<View>
  <CredentialOverview
    claimsToRender={
      Object {
        "Contact": Array [
          Object {
            "claimData": Object {
              "email": "",
            },
            "credentialType": "Email",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "keyboardType": "email-address",
            "subject": "",
          },
          Object {
            "claimData": Object {
              "telephone": "",
            },
            "credentialType": "Mobile Phone",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "keyboardType": "phone-pad",
            "subject": "",
          },
          Object {
            "claimData": Object {
              "addressLine1": "",
              "addressLine2": "",
              "city": "",
              "country": "",
              "postalCode": "",
            },
            "credentialType": "Postal Address",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "subject": "",
          },
        ],
        "Other": Array [
          Object {
            "claimData": Object {
              "random": "thing",
              "something": "else",
            },
            "credentialType": "IdCard",
            "expires": 2019-08-10T00:00:00.000Z,
            "id": "1",
            "issuer": Object {
              "did": "123",
            },
            "subject": "123",
          },
          Object {
            "claimData": Object {
              "random": "thing",
              "something": "else",
            },
            "credentialType": "Passport",
            "id": "2",
            "issuer": Object {
              "did": "123",
            },
            "subject": "123",
          },
          Object {
            "claimData": Object {
              "documentNumber": "12345",
              "random": "thing",
              "something": "else",
            },
            "credentialType": "IdCard",
            "expires": 2019-07-04T00:00:00.000Z,
            "id": "2",
            "issuer": Object {
              "did": "123",
            },
            "subject": "123",
          },
        ],
        "Personal": Array [
          Object {
            "claimData": Object {
              "familyName": "",
              "givenName": "",
            },
            "credentialType": "Name",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "subject": "",
          },
        ],
      }
    }
    did=""
    onEdit={[MockFunction]}
  />
</View>
`;

exports[`Claims container correctly changes scanning to true when qr code scanner is started 1`] = `
<View>
  <CredentialOverview
    claimsToRender={
      Object {
        "Contact": Array [
          Object {
            "claimData": Object {
              "email": "",
            },
            "credentialType": "Email",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "keyboardType": "email-address",
            "subject": "",
          },
          Object {
            "claimData": Object {
              "telephone": "",
            },
            "credentialType": "Mobile Phone",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "keyboardType": "phone-pad",
            "subject": "",
          },
          Object {
            "claimData": Object {
              "addressLine1": "",
              "addressLine2": "",
              "city": "",
              "country": "",
              "postalCode": "",
            },
            "credentialType": "Postal Address",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "subject": "",
          },
        ],
        "Other": Array [
          Object {
            "claimData": Object {
              "random": "thing",
              "something": "else",
            },
            "credentialType": "IdCard",
            "expires": 2019-08-10T00:00:00.000Z,
            "id": "1",
            "issuer": Object {
              "did": "123",
            },
            "subject": "123",
          },
          Object {
            "claimData": Object {
              "random": "thing",
              "something": "else",
            },
            "credentialType": "Passport",
            "id": "2",
            "issuer": Object {
              "did": "123",
            },
            "subject": "123",
          },
          Object {
            "claimData": Object {
              "documentNumber": "12345",
              "random": "thing",
              "something": "else",
            },
            "credentialType": "IdCard",
            "expires": 2019-07-04T00:00:00.000Z,
            "id": "2",
            "issuer": Object {
              "did": "123",
            },
            "subject": "123",
          },
        ],
        "Personal": Array [
          Object {
            "claimData": Object {
              "familyName": "",
              "givenName": "",
            },
            "credentialType": "Name",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "subject": "",
          },
        ],
      }
    }
    did=""
    onEdit={[MockFunction]}
  />
</View>
`;

exports[`Claims container mounts correctly and matches the snapshot 1`] = `
<View>
  <CredentialOverview
    claimsToRender={
      Object {
        "Contact": Array [
          Object {
            "claimData": Object {
              "email": "",
            },
            "credentialType": "Email",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "keyboardType": "email-address",
            "subject": "",
          },
          Object {
            "claimData": Object {
              "telephone": "",
            },
            "credentialType": "Mobile Phone",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "keyboardType": "phone-pad",
            "subject": "",
          },
          Object {
            "claimData": Object {
              "addressLine1": "",
              "addressLine2": "",
              "city": "",
              "country": "",
              "postalCode": "",
            },
            "credentialType": "Postal Address",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "subject": "",
          },
        ],
        "Other": Array [
          Object {
            "claimData": Object {
              "random": "thing",
              "something": "else",
            },
            "credentialType": "IdCard",
            "expires": 2019-08-10T00:00:00.000Z,
            "id": "1",
            "issuer": Object {
              "did": "123",
            },
            "subject": "123",
          },
          Object {
            "claimData": Object {
              "random": "thing",
              "something": "else",
            },
            "credentialType": "Passport",
            "id": "2",
            "issuer": Object {
              "did": "123",
            },
            "subject": "123",
          },
          Object {
            "claimData": Object {
              "documentNumber": "12345",
              "random": "thing",
              "something": "else",
            },
            "credentialType": "IdCard",
            "expires": 2019-07-04T00:00:00.000Z,
            "id": "2",
            "issuer": Object {
              "did": "123",
            },
            "subject": "123",
          },
        ],
        "Personal": Array [
          Object {
            "claimData": Object {
              "familyName": "",
              "givenName": "",
            },
            "credentialType": "Name",
            "id": "",
            "issuer": Object {
              "did": "",
            },
            "subject": "",
          },
        ],
      }
    }
    did=""
    onEdit={[MockFunction]}
  />
</View>
`;
